service/counter/message.go

```diff
@@ -34,10 +35,23 @@ type messageCounterImpl struct {
 // NewMessageCounter 全メッセージ数カウンタを生成します
 func NewMessageCounter(db *gorm.DB, hub *hub.Hub) (MessageCounter, error) {
 	counter := &messageCounterImpl{}
-	if err := db.Unscoped().Model(&model.Message{}).Count(&counter.count).Error; err != nil {
-		return nil, fmt.Errorf("failed to load total messages count: %w", err)
-	}
-	messagesCounter.Add(float64(counter.count))
+
+	go func() {
+		if err := db.Unscoped().Model(&model.Message{}).Count(&counter.count).Error; err != nil {
+			panic(err)
+		}
+
+		once.Do(func() {
+			// Initialize messagesCounter
+			// /api/metrics will not provide traq_messages_count_total until here
+			messagesCounter = promauto.NewCounter(prometheus.CounterOpts{
+				Namespace: "traq",
+				Name:      "messages_count_total",
+			})
+			messagesCounter.Add(float64(counter.count))
+		})
+	}()
+
 	go func() {
 		for range hub.Subscribe(1, event.MessageCreated).Receiver {
```

Comment:
初期化チェックを追加
```suggestion
	go func() {
		<-initDone
		for range hub.Subscribe(1, event.MessageCreated).Receiver {
```

以下の様に、Get() の所にも追加するべき。
```
func (c *messageCounterImpl) Get() int64 {
	<-initDone
	c.RLock()
	defer c.RUnlock()
	return c.count
}
```